cli/azd/pkg/apphost/generate.go

```diff
@@ -632,20 +657,34 @@ func (b *infraGenerator) addServiceBus(name string, queues, topics *[]string) {
 }
 
 func (b *infraGenerator) addInputParameter(name string, comp *Resource) error {
+	input, err := InputParameter(name, comp)
+	if err != nil {
+		return fmt.Errorf("resolving input for parameter %s: %w", name, err)
+	}
+
+	if input == nil {
+		// no inputs in the value, nothing to do
+		return nil
+	}
+
+	b.bicepContext.InputParameters[name] = *input
+	return nil
+}
+
+// InputParameter gets the Input from a parameter. If the parameter does not have an input, it returns nil.
```

Comment:
A parameter can have a fixed value (not an inputParameter) or be linked to one of its inputs.

If linked to `inputs`, azd validates the inputs are referencing the same resource, otherwise there is an error because azd can't resolve inputs from a different resource (like one parameter referencing other parameters' inputs).

When a caller invokes `InputParameter(foo, resource)` and the result is `nil, nil`, the caller knows the parameter is using a fixed value:

![image](https://github.com/Azure/azure-dev/assets/24213737/188f8d7b-234c-4ad8-89b6-cc45c30ba120)

The caller doesn't need to call `HasInputs` and `ValidateInputs`.

For this use case, the `project` package is doing manifest evaluations while the dotnet importer is generating the list of services, so I tried to export only one function from the `apphost` package which was useful enough for the project package to evaluate parameter's logic without knowing too much about the parameters' rules